src/consensus/versionbits.h

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2009-2015 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_VERSIONBITS_H
+#define BITCOIN_VERSIONBITS_H
+
+#include <map>
+#include <set>
+#include <stdexcept>
+#include <stdint.h>
+
+namespace Consensus
+{
+
+const int           NO_RULE      = -1;
+
+namespace VersionBits
+{
+
+const int           VERSION_HIGH_BITS   = 0x20000000;
+const int           VERSION_BITS_MASK   = 0x1fffffff;
+const char          MIN_BIT             = 0;
+const char          MAX_BIT             = 28;
+
+enum RuleState { UNDEFINED, DEFINED, LOCKED_IN, ACTIVE, FAILED };
+
+const char* GetRuleStateText(int ruleState, bool bUseCaps = false);
+
+bool UsesVersionBits(int nVersion);
+
+class SoftFork
+{
+public:
+    SoftFork(int bit, int rule, int threshold, uint32_t deployTime, uint32_t expireTime) :
+        m_bit(bit),
+        m_rule(rule),
+        m_threshold(threshold),
+        m_deployTime(deployTime),
+        m_expireTime(expireTime) { }
+
+    int         GetBit() const { return m_bit; }
+    int         GetRule() const { return m_rule; }
+    int         GetThreshold() const { return m_threshold; }
+    uint32_t    GetDeployTime() const { return m_deployTime; }
+    uint32_t    GetExpireTime() const { return m_expireTime; }
+
+private:
+    int m_bit;
+    int m_rule;
+    int m_threshold;
```

Comment:
To be clear, whether the threshold is per softfork or per chain, it is in Consensus::Params (just once or repeated per softfork). If we only use one per chain (the same for all softforks), we are wasting space, although it is true that not much (even after hundreds of deployments).